### PR TITLE
fix(helm): add --devel flag to allow dev releases again

### LINKED ISSUE #2407: Cannot install/fetch latest chart if 'pre-release' version
Affected helm version: `v2.4.1`

### Issue

`helm [install | fetch] <chart repo>/<chart>` fails if latest `<chart>` in `<chart repo>` has a  "pre-release" version -- that is, contains additional values beyond, say `v0.1.0`, such as `v0.1.0-alpha.1` or `v0.1.0-dev-sha.abc1234`.

### Details

Working in `v2.3.1`:
```
$ helm version
Client: &version.Version{SemVer:"v2.3.1", GitCommit:"32562a3040bb5ca690339b9840b6f60f8ce25da4", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.3.1", GitCommit:"32562a3040bb5ca690339b9840b6f60f8ce25da4", GitTreeState:"clean"}

$ helm search -r workflow-dev
NAME                     	VERSION                               	DESCRIPTION
workflow-dev/workflow    	v2.13.1-dev-20170504190907-sha.3d6bbf9	Deis Workflow

$ helm fetch workflow-dev/workflow
$ ls
workflow-v2.13.1-dev-20170504190907-sha.3d6bbf9.tgz
```

Not working in `v2.4.1`:

```
$ helm version
Client: &version.Version{SemVer:"v2.4.1", GitCommit:"46d9ea82e2c925186e1fc620a8320ce1314cbb02", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.4.1", GitCommit:"46d9ea82e2c925186e1fc620a8320ce1314cbb02", GitTreeState:"clean"}

$ helm search -r workflow-dev
NAME                     	VERSION                               	DESCRIPTION
workflow-dev/workflow    	v2.13.1-dev-20170504190907-sha.3d6bbf9	Deis Workflow
...

$ helm fetch workflow-dev/workflow
Error: chart "workflow" not found in workflow-dev index. (try 'helm repo update'). No chart version found for workflow-
```

However, if the latest chart in the repo has an 'official' release  version, they are successfully fetched in `v2.4.1`:

```
 $ helm search -r jenkins
NAME          	VERSION	DESCRIPTION
stable/jenkins	0.6.2  	Open source continuous integration server. It s..

 $ helm fetch stable/jenkins

 $ ls
jenkins-0.6.2.tgz
```

### Similar/Related Issues

https://github.com/kubernetes/charts/issues/999 was filed specifically for the `stable/traefik` chart.  

Indeed, as we see, this chart happens to have a bit of extra info on the version string (full version: `1.2.1-a`, therefore corroborating this issue's claim that the added info (normally considered "pre-release") is breaking install/fetch:

```
$ helm search -r traefik
NAME          	VERSION	DESCRIPTION
stable/traefik	1.2.1-a	A Traefik based Kubernetes ingress controller w...

$ helm fetch stable/traefik
Error: chart "traefik" not found in stable index. (try 'helm repo update'). No chart version found for traefik-
```

### Caveat

However, it is important to note (and maybe useful in debugging/pinpointing the issue), if the "pre-release" chart version is explicitly specified, it can be installed/fetched with no issue.  Continuing from above:

```
$ helm fetch stable/traefik --version 1.2.1-a

$ ls
traefik-1.2.1-a.tgz
```

### PR DESCRIPTION
**Pull request by technosophos**
2.4.0 stopped installing development versions by default. This allows
the user to re-enable that support by setting --devel.

Closes #2407

### CODE REVIEW

**Comment by technosophos** [comment]
I put this on a patch branch because this appears to be causing headaches for production users.

**Comment by adamreese** [comment]
need to generate docs

File: docs/helm/helm.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ Environment:
 
 ```
       --debug                     enable verbose output
-      --home string               location of your Helm config. Overrides $HELM_HOME (default "~/.helm")
+      --home string               location of your Helm config. Overrides $HELM_HOME (default "~/HelmHome")
```
Comment (adamreese) [inline docs/helm/helm.md]: looks like `HELM_HOME` is set during docs gen?

File: cmd/helm/fetch.go [hunk 2]
```diff
@@ -61,6 +61,8 @@ type fetchCmd struct {
 	keyFile  string
 	caFile   string
 
+	devel bool
```
Comment (adamreese) [inline cmd/helm/fetch.go]: devil 😈

Comment (technosophos) [inline cmd/helm/fetch.go]: I wish we could use --👿 as a flag.

**Review by adamreese — APPROVED** [review]

**Pull request merged**